crates/edr_provider/src/data.rs

```diff
@@ -1895,7 +1908,8 @@ impl<LoggerErrorT: Debug> ProviderData<LoggerErrorT> {
                 .or_else(|| Some(self.parent_beacon_block_root_generator.next_value()));
         }
 
-        let mut inspector = EvmInspector::default();
+        // TODO avoid clone
```

Comment:
Yeah we can't use a reference unfortunately, because it'd prevent mutable borrows of `ProviderData` later. I improved this by wrapping it in an `Arc` so that it's cheaper to clone: [Make call override clone cheaper](https://github.com/NomicFoundation/hardhat/pull/4899/commits/9df9a1ac2988be64b5f7001be23575e07c397939)